### PR TITLE
Framework: Remove sitesList from Pages view

### PR DESCRIPTION
**Pull request by obenland**
This PR aims to help facilitate progress in removing `sitesList` from Calypso. There wasn't a concrete example to point to in last week's framework hangout, so I'm hoping this could be used as one in the next hangout.

It removes `sitesList` from the Pages menu, and in doing so fixed a bug where the `isFrontPage` check didn't return an accurate result.

Before:
screen shot 2016-12-23 at 10 12 54 am

After:
screen shot 2016-12-23 at 10 13 06 am

To test, go to [/pages](http://calypso.localhost:3000/pages/) to verify pages of visible sites are being loaded, as well as site data like site icon and url. Then select a site and view Pages for that site.

I'll be AFK until 1/8, so if this gets approved during that time, please feel free to merge and not wait for me.

See #8726.

### CODE REVIEW

File: client/my-sites/pages/main.jsx [hunk 1]
```diff
@@ -68,6 +72,7 @@ module.exports = React.createClass( {
 		};
 		return (
 			<Main classname="pages">
+				<QuerySites allSites={ true } />
```
Comment (obenland) [inline client/my-sites/pages/main.jsx]: Not sure if this is better or querying every site separately in `<Page />`

Comment (gwwar) [inline client/my-sites/pages/main.jsx]: Thanks @obenland but we'll want to hold off on this PR until we tackle all selected site usages. 

If you look at https://github.com/Automattic/wp-calypso/projects/3 we haven't found a good way of avoiding data duplication between sites list and redux yet.

**Comment by gwwar** [comment]
Going to punt this for now, until we're unblocked @obenland

**Comment by mtias** [comment]
@gwwar @obenland are we now unblocked?

**Comment by gwwar** [comment]
@mtias with the layout changes, we should be unblocked now 👍

**Comment by obenland** [comment]
Fixed in #13317.

**Comment by gwwar** [comment]
Oh thanks for 🔍  @obenland !

**Pull request closed without merging**